docs: organize tvm documentation (#39)
eduNEXT/tvm: docs/TutorEnvironmentManager.rst

Tutor Environment Manager
#########################

Tutor environment manager allows you to create one project and set one version for it.

You will be able to continue using the `tutor version manager`_ commands in your project:

- tvm list
- tvm plugins list

Creating a project
------------------

> To create a new project you should call ``tvm project init``, this configures a new tvm project in the current path and set your-project as ``TUTOR_ROOT`` and ``TUTOR_PLUGINS_ROOT``.  

Reviewer: Maybe we need to specify that TUTOR_ROOT and TUTOR_PLUGINS_ROOT must be absolute paths

Author: I agree but adding in version manager, configuring tvm variables section, where describe how to change it and because project configures it automatically. 

Let me know your opinion.

> You can create different projects with the same tutor version, you have 2 options:

Reviewer: maybe we need to say that if you don't write a name, TVM will create a project in the current fonder and with the name of the current folder, however, if you define a name TVM will create a new folder with that name and in that folder will be created the project.

- Use the global active version.

.. code-block:: bash  
    
    # Using the global active version
    
    mkdir your-project && cd your-project
    tvm project init
    
- Use a specific installed version, this creates a new folder with the ``project-name`` to host it.

.. code-block:: bash  
    
    # Specify the version
    
    tvm project init project-name v14.0.0
    

Activate / Deactivate a project
-------------------------------

> To activate a project you should run

Reviewer: Maybe we need to specify that this command will active a virtualenv with python, pip, and tutor.

.. code-block:: bash
    
    source .tvm/bin/activate
    
    # Verify your environment is active [version@project]
    # That shoud look like: 
    
    [v14.0.0@project-name] os:~/project-name$
    
    
To disable a project you should run ``tvmoff``

If you also enable another environment like python virtualenv, you need to deactivate each virtual environment in order, that's means:

.. code-block:: bash
    
    # (venv) [v12.2.0@project-name] ->
    deactivate
    tvmoff

    # [v12.2.0@project-name] (venv) ->
    tvmoff
    deactivate
    
**Note:** If you forgot to deactivate it in order, you will need to restart your shell terminal.

Check your projects
--------------------

Use ``tvm list`` to check the projects that had been created.

.. image:: images/tvm_list_project.png

Manage a plugin in your project
-------------------------------

**Install**

> To install a tutor plugin in the current tutor version you should use ``pip install <PLUGIN_NAME/or/PLUGIN_LOCAL_PATH/or/PLUGIN_REPO>``

Reviewer: please add the command `tvm plugins install` too.

.. code-block:: bash
    
    pip install tutor-plugin
    pip install /home/user/tutor-plugin
    pip install git+https://github.com/user/tutor-plugin@vx.x.x
    
    
**List**

> List installed plugins with ``tvm plugins list`` or ``tutor plugins list``.

Reviewer: Please, clarify the difference between both.

Tutor Environment Manager
#########################

Tutor environment manager allows you to create one project and set one version for it.

> You will be able to continue using the `tutor version manager`_ commands in your project:
> 
> - tvm list
> - tvm plugins list

Reviewer 2: It seems to me that this information should be a note and not have such a high priority.

```suggestion
**Note:** You can continue using the `tutor version manager`_ commands in your project. Example: `tvm list`, `tvm plugins list`.
```

Creating a project
------------------

To create a new project you should call ``tvm project init``, this configures a new tvm project in the current path and set your-project as ``TUTOR_ROOT`` and ``TUTOR_PLUGINS_ROOT``.  

You can create different projects with the same tutor version, you have 2 options:

**Note:** You could use ``tvm project init project-name``, now TVM will create a new folder with that name and in this folder host the project.

If ``project-name`` isn't specified TVM will create a project in the current folder and configure it with that name.

- Use the global active version.

.. code-block:: bash  
    
    # Using the global active version
    

>     mkdir your-project && cd your-project
>     tvm project init

Reviewer 2: ```suggestion
    tvm project init project-name
```
With that command the folder is already created.

Everything else from the environment manager seems fine to me.
